test/test_baseparser.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: false
+
+require 'rexml/parsers/baseparser'
+
+module REXMLTests
+  class BaseParserTester < Test::Unit::TestCase
+    include REXML
+
+    N_ELEMENTS = 100
+    N_STRING = 'a' * 50000
+    def build_xml(n_elements)
+      xml = '<?xml version="1.0"?><root>'
+
+      n_elements.times do |i|
+        xml << '<child >'
+        xml <<  N_STRING
+        xml <<  '</child>'
+      end
+      xml << '</root>'
+    end
+
+    def test_parse_large_xml
+      xml = build_xml(N_ELEMENTS)
+      parser = REXML::Parsers::BaseParser.new(xml)
+      while parser.has_next?
+        parser.pull
+        assert_compare REXML::Source::SCANNER_RESET_SIZE + N_STRING.size, ">", parser.position
```

Comment:
!!! Can we access `SCANNER_REST_SIZE` via ` REXML::Source` even when we use `private_constant`!?

Reply:
Yes.
```
    20| require 'debug'
    21|     def test_parse_large_xml
    22|       xml = build_xml(N_ELEMENTS)
    23|       parser = REXML::Parsers::BaseParser.new(xml)
=>  24|       debugger
    25|       while parser.has_next?
    26|         parser.pull
    27|         assert_compare REXML::Source::SCANNER_RESET_SIZE + N_STRING.size, ">", parser.position
    28|       end
=>#0	REXMLTests::BaseParserTester#test_parse_large_xml at ~/ghq/github.com/naitoh/rexml/test/parser/test_base.rb:24
  #1	Test::Unit::TestCase#run_test at ~/.rbenv/versions/3.3.2/lib/ruby/gems/3.3.0/gems/test-unit-3.6.1/lib/test/unit/testcase.rb:871
  # and 28 frames (use `bt' command for all frames)
(ruby) puts REXML::Source::SCANNER_RESET_SIZE
100000
nil

```

Reply:
Oh... It seems that we should remove `include Private` (in a separated PR)...

Reply:
Yes...